storage/container.go

```diff
@@ -37,18 +41,60 @@ func (c *Container) GetURL() string {
 	return c.bsc.client.getEndpoint(blobServiceName, pathForResource(container, ""), nil)
 }
 
-func (c *Container) GetSASURI(expiry time.Time, permissions string) (string, error) {
-	return c.GetSASURIWithSignedIPAndProtocol(expiry, permissions, "", false)
+// ContainerSASOptions are options to construct a container SAS
+// URI.
+// See https://docs.microsoft.com/en-us/rest/api/storageservices/constructing-a-service-sas
+type ContainerSASOptions struct {
+	ContainerSASPermissions
+	OverrideHeaders
+	SASOptions
 }
 
-func (c *Container) GetSASURIWithSignedIPAndProtocol(expiry time.Time, permissions string, signedIPRange string, HTTPSOnly bool) (string, error) {
+// ContainerSASPermissions includes the available permissions for
+// a container SAS URI.
+type ContainerSASPermissions struct {
```

Comment:
Seems like this could be a superset of BlobSASPermissions (or perhaps some other "base" type) to avoid duplication.